core/controllers/collection_editor_test.py

```diff
@@ -107,8 +107,8 @@ def test_editable_collection_handler_get(self):
         response = self.testapp.get(
             '%s/%s' % (
                 feconf.COLLECTION_EDITOR_DATA_URL_PREFIX,
-                self.COLLECTION_ID))
-        self.assertEqual(response.status_int, 302)
+                self.COLLECTION_ID), expect_errors=True)
```

Comment:
I don't understand why your PR is causing the underlying behaviour here to change. Conceptually this PR is supposed to be just a renaming, so this looks like an unintended consequence.

Can you dig into this and find out why it changed? Tests are supposed to protect against this sort of thing; when a test fails, it may be due to an error in the newly-introduced code.

Reply:
@seanlip Isn't this an expected behaviour? This test checks that non-editors cannot access the editor data handler due to them not being whitelisted. Hence if we dont add `expect_errors` there's an app error of not being authorised.

Reply:
But why did your PR in particular cause this behaviour to change?

I think 302 typically represents a redirect to the login page.

Reply:
It seems the error occurs when modifying `EditableCollectionDataHandler` in `collection_editor` by adding `GET_HANDLER_ERROR_RETURN_TYPE`. There seems to be some kind of authorisation problem.

Reply:
OK, that's a good start. Can you continue digging into this and tracing what exactly happens, then explaining that in this comment? It's important that we understand what exactly we are modifying, especially with something so crucial as rights management -- we need to get that right.

So, perhaps explain what the flow was like before, and then explain what the flow is now after your modification.

Thanks!

Reply:
Hi @seanlip. The error is occurring in [this](https://github.com/oppia/oppia/blob/develop/core/controllers/base.py#L471) line. For NotLoggedInException its matching if the error_return_type is 'json' or not as there is no payload for GET requests. This error does not occur in other classes as there are either tests for logged in users only or for controllers rendering HTML

Reply:
Ah, ok, thanks for checking! That's a good find.

Looking at that part of the code, I think the logic is not quite correct. Could we modify it so that we return the HTML error page response only if the request is GET and the handler type is HTML, otherwise we return the JSON response?

Reply:
@seanlip Upon changing that piece of code other error pops up like [here](https://github.com/oppia/oppia/blob/develop/core/controllers/profile_test.py#L710) and [here](https://github.com/oppia/oppia/blob/develop/core/domain/acl_decorators_test.py#L1542).
I think the above ^^ is expected and the error that now shows should be changed. If the user is not logged in the error that should be displayed is error 401 and not 302.

Reply:
I agree that 401 errors make sense in both these cases. But they should be JSON responses, right? I'm not sure how this conceptually conflicts with the last comment I made.... could you please explain?

Reply:
done